docs/state.md

```diff
@@ -186,3 +186,47 @@ if err != nil {
 gcsClient, err := storage.New(httpClient)
 
 ```
+
+## Vault (vault://)
+
+As of 1.18, Kops has support for using Vault as state store. It is currently an experimental feature and you have to enable the `VFSVaultSupport` feature flag to enable it.
+
+The goal of the vault store is to be a safe storage for the kops keys and secrets store. It will not work to use this as a kops registry/config store. Among other things, etcd-manager is unable to read VFS control files from vault. Vault also cannot be used as backend for etcd backups.
+
+
+```sh
+export KOPS_FEATURE_FLAGS=VFSVaultSupport
+```
+
+### Node authentication and configuration
+The vault store uses IAM auth to authenticate against the vault server and expects the vault auth plugin to be mounted on `/aws`.
+
+Instructions for configuring your vault server to accept IAM authentication are at https://learn.hashicorp.com/vault/identity-access-management/iam-authentication
+
+To configure kops to use the Vault store, add this to the cluster spec:
+
+```yaml
+spec:
+  secretStore: vault://<vault>:<port>/<kv2 mount>/clusters/<clustername>/secrets
+  keyStore: vault://<vault>:<port>/<kv2 mount/clusters/<clustername>/keys
+```
+
+The `secret` and `key` paths are configurable, but they have to be unique and you should not put other secrets in them.
+
+After launching your cluster you need to add the cluster roles to Vault, binding them to the cluster's IAM identity and granting them access to the appropriate secrets and keys. The nodes will wait until they can authenticate before completing provisioning.
+
+```sh
+vault write auth/aws/role/masters.<clustername> auth_type=iam \
+              bound_iam_principal_arn=arn:aws:iam::<account>:role/masters.<clustername> policies=<my policy> max_ttl=500h
```

Comment:
not knowing too much about how vault works, does kops create the vault policy automatically? or does that need to be externally managed? if so should we provide an example policy to use?

Reply:
Added an example now. Could make kops create policies later, but at least we would not like to let operators have that kind of privilege in vault.